src/Zio.Tests/FileSystems/TestSubFileSystem.cs

```diff
@@ -74,6 +74,43 @@ public void TestWatcher()
         Assert.True(gotChange);
     }
 
+    [SkippableTheory]
+    [InlineData("/test", "/test", "/foo.txt")]
+    [InlineData("/test", "/test", "/~foo.txt")]
+    [InlineData("/test", "/TEST", "/foo.txt")]
+    [InlineData("/test", "/TEST", "/~foo.txt")]
+    [InlineData("/verylongname", "/VERYLONGNAME", "/foo.txt")]
+    [InlineData("/verylongname", "/VERYLONGNAME", "/~foo.txt")]
+    public void TestWatcherCaseSensitive(string physicalDir, string subDir, string filePath)
+    {
+        Skip.IfNot(IsWindows, "This test involves case insensitivity on Windows");
+
+        var physicalFs = GetCommonPhysicalFileSystem();
+        physicalFs.CreateDirectory(physicalDir);
+
+        Assert.True(physicalFs.DirectoryExists(physicalDir));
+        Assert.True(physicalFs.DirectoryExists(subDir));
+
+        var subFs = new SubFileSystem(physicalFs, subDir);
+        var watcher = subFs.Watch("/");
+
+        var gotChange = false;
+        watcher.Created += (sender, args) =>
+        {
+            if (args.FullPath == filePath)
+            {
+                gotChange = true;
+            }
+        };
+
+        watcher.IncludeSubdirectories = true;
+        watcher.EnableRaisingEvents = true;
+
+        physicalFs.WriteAllText($"{physicalDir}{filePath}", "test");
+        Thread.Sleep(100);
```

Comment:
A little bit worried that it could cause issues on heavily loaded CI machines and we would get random CI failures/or even locally. Could you instead create a ManualResetEvent or similar, set it in the created event and wait for it with a timeout here?

Reply:
Sure, shall I do the same in _TestSubFileSystem.TestWatcher_ while I'm at it?

Reply:
Sure, please, if you can, thanks!

Reply:
I suppose I should make another PR with that change for all the other watcher tests that sleep, to avoid unrelated changes here.